File: Example/RtSDKDemo/RtSDKDemo/ViewControllers/Base/GSBaseViewController.m
//  GSBaseViewController.m
//  VodSDKDemo
//  Created by Sheng on 2018/8/3.
//  Copyright © 2018年 Gensee. All rights reserved.
#import "GSBaseViewController.h"
#define UICOLOR16(value) [UIColor colorWithRed:((float)((value & 0xFF0000) >> 16))/255.0 green:((float)((value & 0xFF00) >> 8))/255.0 blue:((float)(value & 0xFF))/255.0 alpha:1.0]
@interface GSBaseViewController ()
@end
@implementation GSBaseViewController
- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    
}
- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    self.automaticallyAdjustsScrollViewInsets = NO;
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}
#pragma mark - extern
- (UILabel *)createElementLabel:(NSString *)tagContent top:(CGFloat)top {
    return [self createElementLabel:tagContent top:top left:15];
}
- (UILabel *)createElementLabel:(NSString *)tagContent top:(CGFloat)top left:(CGFloat)left{
    UILabel *label = [[UILabel alloc] initWithFrame:CGRectMake(left, top, 100, 20)];
    label.text = tagContent;
    label.font = [UIFont systemFontOfSize:14.f];
    [label sizeToFit];
    return label;
}
- (UILabel *)createTagLabel:(NSString *)tagContent top:(CGFloat)top {
    return [self createTagLabel:tagContent top:top left:15];
}
- (UILabel *)createTagLabel:(NSString *)tagContent top:(CGFloat)top left:(CGFloat)left{
    UILabel *label = [[UILabel alloc] initWithFrame:CGRectMake(left, top, 100, 20)];
    label.text = tagContent;
    label.textColor = [UIColor grayColor];
    label.font = [UIFont systemFontOfSize:12.f];
    [label sizeToFit];
    return label;
}
- (UIView *)createWhiteBGViewWithTop:(CGFloat)top itemCount:(NSInteger)count {
    UIView *view         = [[UIView alloc] initWithFrame:CGRectMake(0, top, Width, count * 40.f)];
    view.backgroundColor = [UIColor whiteColor];
    // Top line.
    {
        UIView *line         = [[UIView alloc] initWithFrame:CGRectMake(0, 0, Width, 0.5f)];
        line.backgroundColor = UICOLOR16(0xE8E8E8);
        [view addSubview:line];
    }
    
    // Bottom line.
    {
        UIView *line         = [[UIView alloc] initWithFrame:CGRectMake(0, 0, Width, 0.5f)];
        line.backgroundColor = UICOLOR16(0xE8E8E8);
        line.bottom          = view.height;
        [view addSubview:line];
    }
    
    // Middle lines.
    for (int i = 1; i < count; i++) {
        
        UIView *line         = [[UIView alloc] initWithFrame:CGRectMake(15, i * 40.f - 0.5f, Width - 15.f, 0.5f)];
        line.backgroundColor = UICOLOR16(0xE8E8E8);
        [view addSubview:line];
    }
    return view;
}
@end

File: Example/RtSDKDemo/GSBaseViewController.swift
//
//  GSBaseViewController.swift
//  RtSDKDemoSwift
//
//  Created by gensee on 2019/5/24.
//  Copyright © 2019年 gensee. All rights reserved.
//

import UIKit

let Width = UIScreen.main.bounds.size.width
let Height = UIScreen.main.bounds.size.height
let StatusBarHeight = 20
let NavigationBarHeight = 44
let TabbarHeight = 49
let StatusBarAndNavigationBarHeight = 64
let iPhone4_4s = (Width == 375 && Height == 667 ? true : false)
let iPhoneX = (Width == 375 && Height == 812 ? true : false)

class GSBaseViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = UIColor.white
        self.automaticallyAdjustsScrollViewInsets = true;
        // Do any additional setup after loading the view.
    }
    
    func createTagLabel(tagContent:String,top:CGFloat) -> UILabel {
        return self.createTagLabel(tagContent: tagContent, top: top, left: 15);
    }
    
    func createTagLabel(tagContent:String,top:CGFloat,left:CGFloat) -> UILabel {
        let label = UILabel.init(frame: CGRect.init(x: left, y: top, width: 100, height: 20))
        label.text = tagContent;
        label.textColor = UIColor.gray;
        label.font = UIFont.systemFont(ofSize: 12);
        label.sizeToFit();
        return label;
    }
    
    func createWhiteBGView(top t:CGFloat,count c:NSInteger) -> UIView {
        let view = UIView.init(frame: CGRect.init(x: CGFloat(0), y: t, width: Width, height: CGFloat(c*40)));
        view.backgroundColor = UIColor.white;
        let topline = UIView.init(frame: CGRect.init(x: 0, y: 0, width: Width, height: 0.5));
        topline.backgroundColor = #colorLiteral(red: 0.6000000238, green: 0.6000000238, blue: 0.6000000238, alpha: 1)
        view.addSubview(topline)
        let botline = UIView.init(frame: CGRect.init(x: 0, y: 0, width: Width, height: 0.5));
        botline.backgroundColor = #colorLiteral(red: 0.6000000238, green: 0.6000000238, blue: 0.6000000238, alpha: 1)
        botline.bottom = view.height
        view.addSubview(botline)
        
        for i in 1...c {
            let topline = UIView.init(frame: CGRect.init(x: Double(15), y: Double(i * 40) - 0.5, width: Double(Width - 15), height: 0.5));
            topline.backgroundColor = #colorLiteral(red: 0.6000000238, green: 0.6000000238, blue: 0.6000000238, alpha: 1)
            view.addSubview(topline)
        }
        return view;
    }
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
    */

}
